packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol

```diff
@@ -562,7 +567,18 @@ contract RoutersFacet is BaseConnextFacet {
     if (_amount == 0) revert RoutersFacet__addLiquidityForRouter_amountIsZero();
 
     // Get the canonical asset ID from the representation.
-    (TokenId memory canonical, bytes32 key) = _getApprovedCanonicalId(_local);
+    // NOTE: not using `_getApprovedCanonicalId` because candidate can *only* be local
+    TokenId memory canonical = s.representationToCanonical[_local];
+    if (canonical.domain == 0) {
```

Comment:
Here we are assuming that when `canonical.domain == 0`, `canonical.id` is also not set. Perhaps we should check this.

Reply:
added! https://github.com/connext/nxtp/pull/2489/commits/1afe00a3e5bddbf0424cf954b1f77a807b1aa6d5